src/core_read.cpp

```diff
@@ -50,15 +50,14 @@ CScript ParseScript(const std::string& s)
 
     for (std::vector<std::string>::const_iterator w = words.begin(); w != words.end(); ++w)
     {
+        int64_t n;
         if (w->empty())
         {
             // Empty string, ignore. (boost::split given '' will return one word)
         }
-        else if (std::all_of(w->begin(), w->end(), ::IsDigit) ||
-            (w->front() == '-' && w->size() > 1 && std::all_of(w->begin()+1, w->end(), ::IsDigit)))
+        else if (ParseInt64(*w, &n))
```

Comment:
In commit "refactor: Replace use of atoi64 in core_read" (e4aa1f77b16c248488cc88b661bed69168c0f9e5)

Parsing here seems a little looser than before, because `ParseScript` will now accept signed numbers beginning with `+`, while it previously only accepted numbers beginning with `-`. I'm assuming this is reasonable behavior. `ParseScript` appears to only actually be called in tests and by `bitcoin-tx`.

Reply:
`ParseIntXX` accepts signed numbers beginning with `+`? whoops, I don't think this was intentional.